samples/subsys/usb/console/src/main.c

```diff
@@ -6,29 +6,30 @@
 
 #include <zephyr.h>
 #include <sys/printk.h>
-#include <sys/util.h>
 #include <string.h>
 #include <usb/usb_device.h>
 #include <drivers/uart.h>
 
+#define CDC_ACM_LABEL DT_LABEL(DT_COMPAT_GET_ANY_STATUS_OKAY(zephyr_cdc_acm_uart))
+
 void main(void)
 {
-	const struct device *dev = device_get_binding(
-			CONFIG_UART_CONSOLE_ON_DEV_NAME);
+	const struct device *dev = DEVICE_DT_GET_ONE(zephyr_cdc_acm_uart);
 	uint32_t dtr = 0;
 
 	if (usb_enable(NULL)) {
 		return;
 	}
 
-	/* Poll if the DTR flag was set, optional */
+	/* Poll if the DTR flag was set */
 	while (!dtr) {
 		uart_line_ctrl_get(dev, UART_LINE_CTRL_DTR, &dtr);
+		/* Give CPU resources to low priority threads. */
+		k_sleep(K_MSEC(100));
 	}
 
-	if (strlen(CONFIG_UART_CONSOLE_ON_DEV_NAME) !=
-	    strlen("CDC_ACM_0") ||
-	    strncmp(CONFIG_UART_CONSOLE_ON_DEV_NAME, "CDC_ACM_0",
+	if (strlen(CONFIG_UART_CONSOLE_ON_DEV_NAME) != strlen(CDC_ACM_LABEL) ||
+	    strncmp(CONFIG_UART_CONSOLE_ON_DEV_NAME, CDC_ACM_LABEL,
```

Comment:
I'm confused about what's going on here. If you're polling DTR on `dev`, then I think the right comparison is that `dev->name` is `CONFIG_UART_CONSOLE_ON_DEV_NAME` instead. Although that still isn't 100% correct since there could be multiple CDC ACM devices, and `dev` is one, but it isn't the console.

Reply:
there is no point in using `strncmp` as `strlen` that is used for limit is called with the same argument

Reply:
but both issues seems historical

Reply:
AFAIK the idea was to catch the error if chosen node and CONFIG_UART_CONSOLE_ON_DEV_NAME has the default value from board's dts. But this is only the case if someone changes this sample to try something, e.g. remove or rename chosen node in samples app.overlay.

I can remove it if desired.

Reply:
> I can remove it if desired.

See above for a suggested solution.